src/option.c

```diff
@@ -296,7 +298,7 @@ static void option_parse_token(char ** argv, int * flag_parsed_next, int * requi
             case (OPTION_OPTIONAL_ARGUMENT):
             case (OPTION_REQUIRED_ARGUMENT):{
               // check if next is an argument
-              if(arg == NULL){
+              if(arg == NULL && replaced_equal != 1){
```

Comment:
I'm not quite sure why this is necessary? It seems to me the fix in L259 takes care of it. e.g. if it isn't NULL, then the actual value shall be used.

Reply:
This whole code is hard to read. Here is my understanding:

L258 and L259 make the argument empty in case no value was provided after '=' (without those two lines \0 end-of-string char was used as value in case no value is provided to the option). 

The 'if' block starting at L301, fetches next argument for the value in case arg is NULL. We don't want to do that as when replaced_equal == 1 as we want to return an error (handled line L312). Anyway I don't think we want to check next argument if '=' is detected (would '--backend.option= 1' be valid?)

A few examples:
1) --backend.option=        (before: value 0, with the patch: "Error, argument missing for option...")
2) --backend.option=1      (before: value 1, with the patch: value 1)
3) --backend.option= 1     (before: value 0 and error on unknown 1 argument, with the patch: "Error, argument missing for option...")

Reply:
Yes, patch seems good.